### PR TITLE
API parameters to set Envmap/BRDF LUT sample counts

### PR DESCRIPTION
**Pull request by saphiresouldier**
This PR turns the hardcoded sampling counts for PrecomputedEnvMap/IrradianceMap/BRDFLUT into parameters than can be set from outside.

### CODE REVIEW

File: GLTF_PBR_Renderer/interface/GLTF_PBR_Renderer.hpp [hunk 1]
```diff
@@ -109,7 +109,8 @@ class GLTF_PBR_Renderer
     GLTF_PBR_Renderer(IRenderDevice*     pDevice,
                       IRenderStateCache* pStateCache,
                       IDeviceContext*    pCtx,
-                      const CreateInfo&  CI);
+                      const CreateInfo&  CI,
+                      int                numBRDFSamples = 512);
```
Comment (TheMostDiligent) [inline GLTF_PBR_Renderer/interface/GLTF_PBR_Renderer.hpp:113]: I suggest adding this parameter to the `CreateInfo` struct.
Please use naming style consistent with the rest of the API: `NumBRDFSamples`
Please use `Uint32` type

File: GLTF_PBR_Renderer/interface/GLTF_PBR_Renderer.hpp [hunk 2]
```diff
@@ -223,7 +224,10 @@ class GLTF_PBR_Renderer
     void PrecomputeCubemaps(IRenderDevice*     pDevice,
                             IRenderStateCache* pStateCache,
                             IDeviceContext*    pCtx,
-                            ITextureView*      pEnvironmentMap);
+                            ITextureView*      pEnvironmentMap,
+                            int                numPhiSamples   = 64,
```
Comment (TheMostDiligent) [inline GLTF_PBR_Renderer/interface/GLTF_PBR_Renderer.hpp:228]: Please use consistent naming style.
Please use `Uint32`.

File: GLTF_PBR_Renderer/interface/GLTF_PBR_Renderer.hpp [hunk 3]
```diff
@@ -223,7 +224,10 @@ class GLTF_PBR_Renderer
...
+                            int                numThetaSamples = 32,
+                            int                optimizeSamples = 1);
```
Comment (TheMostDiligent) [inline GLTF_PBR_Renderer/interface/GLTF_PBR_Renderer.hpp:230]: Please use `bool`

**Review by TheMostDiligent — CHANGES_REQUESTED** [review]

File: GLTF_PBR_Renderer/interface/GLTF_PBR_Renderer.hpp (hunk 1 above)
Comment (saphiresouldier) [reply GLTF_PBR_Renderer/interface/GLTF_PBR_Renderer.hpp]: Done as requested!

**Review by TheMostDiligent — APPROVED** [review]

**Pull request merged**